src/fromager/bootstrapper.py

```diff
@@ -40,6 +41,21 @@
 SeenKey = tuple[NormalizedName, tuple[str, ...], str, typing.Literal["sdist", "wheel"]]
 
 
+@dataclasses.dataclass
+class SourceBuildResult:
+    """Result of building a package from source.
+
+    Used to return multiple values from _build_from_source().
+    """
+
+    wheel_filename: pathlib.Path | None
+    sdist_filename: pathlib.Path | None
+    unpack_dir: pathlib.Path
+    sdist_root_dir: pathlib.Path
+    build_env: build_environment.BuildEnvironment
+    source_url_type: str
```

Comment:
Let's call it `source_type` and use the `SourceType` enum

```suggestion
    source_type: SourceType
```